plugins/integration/lib/events/kIntegrationFlowManager.php

```diff
@@ -40,6 +40,19 @@ public static function addintegrationJob($objectType, $objectId, kIntegrationJob
 	{
 		$partnerId = kCurrentContext::getCurrentPartnerId();
 		
+		$providerType = $data->getProviderType();
+		$integrationProviderObj = IntegrationProviderPlugin::getProvider($providerType);
+		
+		if($integrationProviderObj)
+		{
+			$permissions = $integrationProviderObj->getPermissions($partnerId);

```

Comment:
If you just count the permissions, you better move all the implementation into the provider and call it validatePermissions, or keep it getPermissions but just return the required permission names to be validated here.